### PR TITLE
feature/sunset

### LINKED ISSUE #89: new handler design: sunset header detection
AGS has started [returning sunset headers](https://datatracker.ietf.org/doc/html/rfc8594) on deprecated operations.
We already have a tracing infrastructure thanks to Open Telemetry and create spans that are enriched by each middleware hander.
Implementing a handler that warns on sunset would allow admins and SecOps to provide feedback to their development teams so they can update an application before it gets impacted by APIs retirement.

We could had a new handler that would:
1. start a new span (parent span from the context
1. add a boolean attribute about whether the operation returned a sunset header or not.
1. set the span status to error when a sunset header is present and the represented date is less that a year from now (default value), with the description being the an explicit message saying this API will be inoperative at date X
1. add date attribute to the span when present

This handler can be configured with its dedicated option, the option provide the following fields:
- enabled (boolean), default true
- sunsetAsErrorDelay (timespan), default one year

We don't need to log the URL or the HTTP verb as attributes as this is already done by the client/other handlers.

### PR DESCRIPTION
**Pull request by baywet**
fixes #89

### CODE REVIEW

**Review by MIchaelMainer — APPROVED** [review]
I really like this feature.

File: middleware/SunsetHandler.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Sunset Handler
+
+## Objectives
+
+Let application operation teams know that a given operation depends on an API which is slated for removal so they can proactively update the application before a breakage occurs.
```
Comment (musale) [inline middleware/SunsetHandler.md:5]: For more context, would adding that this handler uses the Sunset header only to communicate that a resource is slated for removal. In the RFC, a resource might communicate about sunset policy by using the sunset link but that is not handled in the scope of this handler.

Comment (baywet) [reply middleware/SunsetHandler.md:5]: I don't think those are mutually exclusive. From reading the RFC my understanding is that the link can be provided even in a deprecation scenario.
And there's nothing in the RFC that'd allow us to mechanically parse the sunset "type" either. So from a client middleware perspective, I think we need to treat them as equivalent.
Now, you bring a good point, maybe we should also parse the link header and add it to the event attributes when present so admins have a hint to where start looking at. I'll make that addition.

Comment (baywet) [reply middleware/SunsetHandler.md:5]: (sorry apparently the pr was on auto-merge and when I pushed the resolution of the conflict, it got merged)

Comment (musale) [reply middleware/SunsetHandler.md:5]: Parsing the link and adding it as an event is a good idea. So, either header or link.

**Pull request merged**